npm install fails: 'TS2349: This expression is not callable'

<!--
Thank you for reporting an issue.

This issue tracker is for bugs found within the JavaScript implementation of libp2p.

If you are asking a question about how to use libp2p, please ask on https://discuss.libp2p.io

Otherwise please fill in as much of the template below as possible.
-->

- **Version**:
<!--
Check package.json version
-->
0.33.0

- **Platform**:
<!--
Output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows). If using in a Browser, please share the browser version as well
-->
Linux trout-dev 5.11.0-34-generic #36~20.04.1-Ubuntu SMP Fri Aug 27 08:06:32 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

- **Subsystem**:
<!--
If known, please specify affected core module name (e.g Dialer, Pubsub, Relay etc)
-->
N/A

#### Severity:
<!--
One of following:
  Critical - System crash, application panic.
  High - The main functionality of the application does not work, API breakage, repo format breakage, etc.
  Medium - A non-essential functionality does not work, performance issues, etc.
  Low - An optional functionality does not work.
  Very Low - Translation or documentation mistake. Something that won't give anyone a bad day.
-->
High

#### Description:
<!--
- What you did
- What happened
- What you expected to happen
-->
Trying to install the `master` branch (commit a335fda) as well as the latest release (v0.33.0) commit, both fail on `npm install`

Node version: v14.17.6
npm version: 7.23.0
OS: Ubuntu 20.4

#### Steps to reproduce the error:
<!--
If possible, please provide code that demonstrates the problem, keeping it as simple and free of external dependencies as you are able
-->
- `git clone` the repository.
- (optional) `git checkout v0.33.0`
- Run `npm install` and notice the following error:

```
npm ERR! code 1
npm ERR! path /home/trout/tmp/js-libp2p
npm ERR! command failed
npm ERR! command sh -c aegir build --no-bundle
npm ERR! [06:03:45] Clean ./dist [started]
npm ERR! [06:03:45] Clean ./dist [completed]
npm ERR! [06:03:45] Generate types [started]
npm ERR! src/nat-manager.js(118,11): error TS2349: This expression is not callable.
npm ERR!   Type 'typeof import("/home/trout/tmp/js-libp2p/node_modules/private-ip/lib/index")' has no call signatures.
npm ERR! [06:03:55] Generate types [failed]
npm ERR! [06:03:55] → Command failed with exit code 1: tsc --build /home/trout/tmp/js-libp2p/tsconfig-types.aegir.json
npm ERR! Command failed with exit code 1: tsc --build /home/trout/tmp/js-libp2p/tsconfig-types.aegir.json

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/trout/.npm/_logs/2021-11-10T14_04_00_258Z-debug.log

```

--- a/src/nat-manager.js
+++ b/src/nat-manager.js
@@ -10,7 +10,6 @@ const log = Object.assign(debug('libp2p:nat'), {
 })
 const { isBrowser } = require('wherearewe')
 const retry = require('p-retry')
-// @ts-ignore private-api does not export types
 const isPrivateIp = require('private-ip')
 const pkg = require('../package.json')
 const errcode = require('err-code')
@@ -115,6 +114,7 @@ class NatManager {
       const client = this._getClient()
       const publicIp = this._externalIp || await client.externalIp()
 
+      // @ts-ignore isPrivate has no call signatures
       if (isPrivateIp(publicIp)) {
         throw new Error(`${publicIp} is private - please set config.nat.externalIp to an externally routable IP or ensure you are not behind a double NAT`)
       }

